ui/app/appointments/controllers/manage/calendar/appointmentsDayCalendarController.js

```diff
@@ -54,14 +54,21 @@ angular.module('bahmni.appointments')
                     uiCalendarConfig.calendars[calendar].fullCalendar('changeView', view);
                 };
 
-                var isUserPrivilegedToCreateAppointment = function () {
+                function isOwnAppointmentUserPrivilegedToCreateAppointment (privilege, provider) {
+                    const NO_PROVIDER_UUID = 'no-provider-uuid';
+                    return (privilege.name === Bahmni.Appointments.Constants.privilegeOwnAppointments &&
+                        (provider.uuid === NO_PROVIDER_UUID || provider.uuid === $rootScope.currentProvider.uuid));
+                }
+
+                var isUserPrivilegedToCreateAppointment = function (provider) {
```

Comment:
Should this not be simple check of whether user has "Manage Appointment" or "Self Appointment" privilege? How is this check different?

Reply:
A provider with manage own appointment privilege will be able to create the appointments for a no provider(appointment with no provider). That is why this check is there. If we click on the slot of a no provider in calendar view to create a new appointment, it should allow the user to create the appointment but the same should not be allowed for any other provider.